src/tty.c

```diff
@@ -40,7 +40,13 @@ METAL_CONSTRUCTOR(metal_tty_init) {
  * provide a shim that eats all the characters so we can ensure that everything
  * can link to metal_tty_putc. */
 int nop_putc(int c) __attribute__((section(".text.metal.nop.putc")));
-int nop_putc(int c) { return -1; }
+// Use a customizable NOP hint instruction so that a post-processor parser can
+// look for this instruction, and use the value in a0 as the character to be
+// printed.
+int nop_putc(int c) {
+    asm volatile ("slli x0,a0,0x11");
```

Comment:
Pretty sure we need you to use `__asm__` here in order to comply with `-std=c99`.

Cool trick!

Reply:
Ah, good catch! Fixed!